ra2ce/network/networks_utils.py

```diff
@@ -1539,6 +1539,43 @@ def fraction_flooded(line: LineString, hazard_map: str):
     except Exception as e:
         logging.info("fraction_flooded() {} \n for line {}".format(e, line))
 
+def fraction_flooded_array(line: LineString, raster_array: np.ndarray, affine) -> float:
```

Comment:
Why is this method here if it is only required by `hazard_intersect_builder_for_tif` ? Is it meant to be used as a standalone method outside ra2ce workflows?
With the current context this fits better in the mentioned class as a static (private) method.

In addition, regardless of having it here or there, please add tests to validate the calculation is correctly done. Otherwise this feature will not be merged (or reverted otherwise).